### PR TITLE
Remove deprecated options

### PR DESCRIPTION
**Pull request by spencertransier**
## What does it do?

- Removed deprecated `has_alpha_version`, `project_name`, and `project_root_folder` options from several actions in preparation for the 10.0 Release Toolkit release. 

## Checklist before requesting a review

- [X] Run `bundle exec rubocop` to test for code style violations and recommendations
~- [ ] Add Unit Tests (aka `specs/*_spec.rb`) if applicable~
- [X] Run `bundle exec rspec` to run the whole test suite and ensure all your tests pass
- [X] Make sure you added an entry in [the `CHANGELOG.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/CHANGELOG.md#trunk) to describe your changes under the appropriate existing `###` subsection of the existing `## Trunk` section.
- [X] If applicable, add an entry in [the `MIGRATION.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/MIGRATION.md) to describe how the changes will affect the migration from the previous major version and what the clients will need to change and consider.

### CODE REVIEW

File: lib/fastlane/plugin/wpmreleasetoolkit/helper/deprecated.rb
Comment (spencertransier) [inline lib/fastlane/plugin/wpmreleasetoolkit/helper/deprecated.rb:1]: I decided to just remove this file completely. `deprecated.rb` could be added back later if needed, but I thought it didn't make sense to keep around an empty file indefinitely.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_git_helper.rb [hunk 1]
```diff
@@ -9,8 +9,6 @@ module GitHelper
         # This typically commits the `version.properties` inside root folder or `build.gradle` file
         # inside the project subfolder.
         #
-        # @env PROJECT_ROOT_FOLDER The path to the git root of the project
-        # @env PROJECT_NAME The name of the directory containing the project code (especially containing the `build.gradle` file)
         #
```
Comment (AliSoftware) [inline lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_git_helper.rb]:
```suggestion
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_update_release_notes.rb [hunk 2]
```diff
@@ -41,7 +39,7 @@ def self.available_options
                                        env_name: 'FL_ANDROID_UPDATE_RELEASE_NOTES_FILE_PATH',
                                        description: 'The path to the release notes file to be updated',
                                        type: String,
-                                       default_value: File.join(ENV['PROJECT_ROOT_FOLDER'] || '.', 'RELEASE-NOTES.txt')),
+                                       default_value: File.join('.', 'RELEASE-NOTES.txt')),
```
Comment (AliSoftware) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_update_release_notes.rb]: At this point the `./` path prefix becomes unnecessary.
```suggestion
                                       default_value: 'RELEASE-NOTES.txt'),
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_update_release_notes.rb [hunk 3]
```diff
@@ -41,7 +39,7 @@ def self.available_options
                                        env_name: 'FL_IOS_UPDATE_RELEASE_NOTES_FILE_PATH',
                                        description: 'The path to the release notes file to be updated',
                                        type: String,
-                                       default_value: File.join(ENV['PROJECT_ROOT_FOLDER'] || '.', 'RELEASE-NOTES.txt')),
+                                       default_value: File.join('.', 'RELEASE-NOTES.txt')),
```
Comment (AliSoftware) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_update_release_notes.rb]:
```suggestion
                                       default_value: 'RELEASE-NOTES.txt'),
```

**Review by AliSoftware — APPROVED** [review]

**Pull request merged**